frontend/src/App.tsx

```diff
@@ -14,14 +14,44 @@ import ProtectedRoutesAdminAndOrganizer from "./components/ProtectedRoutesAdminA
 import ProtectedRoutesAdminAndEditor from "./components/ProtectedRoutesAdminAndEditor";
 import {ToastContainer} from "react-toastify";
 import 'react-toastify/dist/ReactToastify.css';
+import PreferredEventCategorySelection from "./components/PreferredEventCategorySelection";
 
 function App() {
 
-    const {login, register, logout, user} = useUser();
-    const {getAllEvents, events} = useEvent();
+    const [areNavLinksVisible, setAreNavLinksVisible] = useState(false);
+    const {user, register, login, logout, updateUserPreferredCategories} = useUser();
+    const {
```

Comment:
why not create an object with all these useStates as props?

Reply:
Could you please describe again what you mean? Currently I see only one useState in my App.tsx

Reply:
I was asking about the option to create an object with props of all the listed params line 24 and following such that the useHook is only this object.